src/SprykerEco/Zed/AvalaraTax/Business/AvalaraTaxBusinessFactory.php

```diff
@@ -0,0 +1,226 @@
+            $this->getMoneyFacade(),
+            $this->getUtilEncodingService(),
+            $this->getCreateTransactionRequestAfterPlugins()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Executor\AvalaraTransactionExecutorInterface
+     */
+    public function createAvalaraTransactionExecutor(): AvalaraTransactionExecutorInterface
+    {
+        return new AvalaraTransactionExecutor(
+            $this->createAvalaraTransactionBuilder(),
+            $this->createAvalaraTransactionRequestMapper(),
+            $this->createAvalaraTransactionResponseMapper(),
+            $this->createAvalaraTransactionLogger(),
+            $this->getUtilEncodingService()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Mapper\AvalaraTransactionRequestMapperInterface
+     */
+    public function createAvalaraTransactionRequestMapper(): AvalaraTransactionRequestMapperInterface
+    {
+        return new AvalaraTransactionRequestMapper(
+            $this->getConfig(),
+            $this->getMoneyFacade(),
+            $this->getCreateTransactionRequestExpanderPlugins()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Mapper\AvalaraTransactionResponseMapperInterface
+     */
+    public function createAvalaraTransactionResponseMapper(): AvalaraTransactionResponseMapperInterface
+    {
+        return new AvalaraTransactionResponseMapper($this->getUtilEncodingService());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Builder\AvalaraTransactionBuilderInterface
+     */
+    public function createAvalaraTransactionBuilder(): AvalaraTransactionBuilderInterface
+    {
+        return new AvalaraTransactionBuilder($this->createTransactionBuilder());
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Business\Logger\AvalaraTransactionLoggerInterface
+     */
+    public function createAvalaraTransactionLogger(): AvalaraTransactionLoggerInterface
+    {
+        return new AvalaraTransactionLogger(
+            $this->getEntityManager(),
+            $this->getStoreFacade()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Dependency\External\AvalaraTaxToTransactionBuilderInterface
+     */
+    public function createTransactionBuilder(): AvalaraTaxToTransactionBuilderInterface
+    {
+        return new AvalaraTaxToAvalaraTransactionBuilderAdapter(
+            $this->createTransactionClient(),
+            $this->getConfig()->getCompanyCode(),
+            (string)AvalaraTaxConfig::AVALARA_TRANSACTION_TYPE_ID_SALES_ORDER,
+            $this->getConfig()->getDefaultCustomerCode()
+        );
+    }
+
+    /**
+     * @return \SprykerEco\Zed\AvalaraTax\Dependency\External\AvalaraTaxToAvalaraTaxClientInterface
+     */
+    public function createTransactionClient(): AvalaraTaxToAvalaraTaxClientInterface
+    {
+        $avaTaxClient = new AvalaraTaxToAvalaraAvaTaxClientAdapter(
```

Comment:
* The adapter has to be created in the dependency provider.
Example: `\Spryker\Service\UtilValidate\UtilValidateDependencyProvider::addEmailValidator`.

Reply:
Resolved.